src/index.rs

```diff
@@ -966,23 +966,27 @@ mod tests {
     }
   }
 
-  struct Context {
+  pub(crate) struct Context {
```

Comment:
This is the test `Context` that before this PR was only used in the tests here in `index`. Now the tests in `export` also use this `Context`, so I made it `pub(crate)` (and changed it a little). Note that for production code, this is still not visible or exported, since the whole `test` module is guarded through `#[cfg(test)]`.

It feels a bit weird to import test helpers from another test module, so I would be up for moving this to a dedicated test helpers module. Although it does work fine and I actually like it, now that I'm used to it. I *do* think these common test helpers should not be duplicated. Any opinions?